44.md

```diff
@@ -0,0 +1,255 @@
+NIP-44
+======
+
+Encrypted Payloads (Versioned)
+------------------------------
+
+`optional` `author:paulmillr` `author:staab`
+
+The NIP introduces a new data format for keypair-based encryption. This NIP is versioned to allow multiple algorithm choices to exist simultaneously.
+
+The encrypted payload is encoded as base64 using the character set defined in RFC4648. The first byte of the payload represents the version. The rest depends on the version.
+
+Currently defined encryption algorithms:
+
+- `0x00` - Reserved
+- `0x01` - XChaCha20 with same key `sha256(parity + ecdh)` per conversation
```

Comment:
Actually the parity byte is part of the output of ecdh for most libraries, so this would put it in twice.

I would put something like `ecdh(private_key, public_key) where the ecdh() function computes the shared point, and then returns the sha256 hash of the x-only component prefixed by the parity byte ('02').`

Reply:
except that the libraries implement ECDSA and not schnorr. There is no parity byte in schnorr.

Reply:
I only understand a little. I thought the parity byte represented the Y coordinate and there were two options since the curve loops back around on itself. And I know that with Schnorr signatures it is always even (2) for some reason.  And I thought DH just did successive multiplications to come up to the same point  g^a * g^b = g^b * g^a and I don't understand where either signature algorithm (DSA or Schnorr) comes in when determining the DH shared point (and does ECDSA even have a shared point?).

All I want is for the NIP to do the default thing that libraries do, rather than a custom thing that requires lower-level operations.